packages/flame/lib/src/effects2/size_effect.dart

```diff
@@ -0,0 +1,46 @@
+import '../../components.dart';
+import '../../extensions.dart';
+import 'component_effect.dart';
+import 'effect_controller.dart';
+
+/// Change the size of a component over time.
+///
+/// The following constructors are provided:
+///
+///   - [SizeEffect.by] will set the size in relation to it's current size;
+///   - [SizeEffect.to] will set the size to the specified size
+///
+/// This effect applies incremental changes to the component's size, and
+/// requires that any other effect or update logic applied to the same component
+/// also used incremental updates.
+class SizeEffect extends ComponentEffect<PositionComponent> {
+  SizeEffect.by(Vector2 offset, EffectController controller)
+      : _offset = offset.clone(),
+        super(controller);
+
+  factory SizeEffect.to(Vector2 targetSize, EffectController controller) =>
+      _SizeToEffect(targetSize, controller);
+
+  Vector2 _offset;
+
+  @override
+  void apply(double progress) {
+    final dProgress = progress - previousProgress;
+    target.size += _offset * dProgress;
```

Comment:
I wonder what would happen if the size becomes negative, and whether there is something we could do about it?
Obviously, reduction in component's size should be allowed, but this could easily drive the size into the negative range (especially if multiple effects are applied), and I wonder if it would break anything?

Reply:
It probably would have some weird consequences, I'll cap it at 0.